File: Source/ObjC/NSKBottomImageLayout.m
//
//  NSKBottomImageLayout.m
//  NSKButtonObjC
//
//  Created by NSSimpleApps on 21.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "NSKBottomImageLayout.h"

@implementation NSKBottomImageLayout

+ (CGRect)titleRectForContentRect:(CGRect)contentRect
                 defaultTitleRect:(CGRect)defaultTitleRect
                        imageRect:(CGRect)imageRect
                  titleEdgeInsets:(UIEdgeInsets)titleEdgeInsets {
    
    if (CGRectIsEmpty(defaultTitleRect)) {
        
        return [super titleRectForContentRect:contentRect
                             defaultTitleRect:defaultTitleRect
                                    imageRect:imageRect
                              titleEdgeInsets:titleEdgeInsets];
        
    } else {
        
        return CGRectMake(CGRectGetMidX(contentRect) - defaultTitleRect.size.width/2 + titleEdgeInsets.left - titleEdgeInsets.right,
                          CGRectGetMinY(contentRect) + titleEdgeInsets.top - titleEdgeInsets.bottom,
                          defaultTitleRect.size.width,
                          defaultTitleRect.size.height);
    }
}

+ (CGRect)imageRectForContentRect:(CGRect)contentRect
                 defaultImageRect:(CGRect)defaultImageRect
                        titleRect:(CGRect)titleRect
                  imageEdgeInsets:(UIEdgeInsets)imageEdgeInsets {
    
    if (CGRectIsEmpty(defaultImageRect)) {
        
        return [super imageRectForContentRect:contentRect
                             defaultImageRect:defaultImageRect
                                    titleRect:titleRect
                              imageEdgeInsets:imageEdgeInsets];
        
    } else {
        
        return CGRectMake(CGRectGetMidX(contentRect) - defaultImageRect.size.width/2 + imageEdgeInsets.left - imageEdgeInsets.right,
                          CGRectGetMaxY(contentRect) - defaultImageRect.size.height + imageEdgeInsets.top - imageEdgeInsets.bottom,
                          defaultImageRect.size.width,
                          defaultImageRect.size.height);
    }
}

@end

File: Source/Swift/NSKBottomImageLayout.swift
//
//  NSKBottomImageLayout.swift
//  NSKButtonSwift
//
//  Created by NSSimpleApps on 24.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit


internal class NSKBottomImageLayout: NSKVerticalImageLayout {
    
    override static func titleRect(forContentRect contentRect: CGRect,
                                  defaultTitleRect: CGRect,
                                  imageRect: CGRect,
                                  titleEdgeInsets: UIEdgeInsets) -> CGRect {
        
        return CGRect(x: contentRect.midX - defaultTitleRect.width/2 + titleEdgeInsets.left - titleEdgeInsets.right,
                      y: contentRect.minY + titleEdgeInsets.top - titleEdgeInsets.bottom,
                      width: defaultTitleRect.width,
                      height: defaultTitleRect.height)
    }
    
    override static func imageRect(forContentRect contentRect: CGRect,
                                  defaultImageRect: CGRect,
                                  titleRect: CGRect,
                                  imageEdgeInsets: UIEdgeInsets) -> CGRect {
        
        return CGRect(x: contentRect.midX - defaultImageRect.width/2 + imageEdgeInsets.left - imageEdgeInsets.right,
                      y: contentRect.maxY - defaultImageRect.height + imageEdgeInsets.top - imageEdgeInsets.bottom,
                      width: defaultImageRect.width,
                      height: defaultImageRect.height)
    }
}
